packages/components/src/client/components/BiliBili.ts

```diff
@@ -51,11 +57,21 @@ export default defineComponent({
         allow:
           "accelerometer; autoplay; clipboard-write; encrypted-media; fullscreen; gyroscope; picture-in-picture",
         style: {
-          height:
-            typeof props.height === "string"
-              ? props.height
-              : `${props.height}px`,
+          width:
+            typeof props.width === "string" ? props.width : `${props.width}px`,
         },
       });
   },
+  mounted() {
+    const bili = this.$refs["bili"] as HTMLIFrameElement;
+
+    bili.style.height =
+      typeof this.height === "string"
+        ? this.height.toLowerCase() == "auto"
+          ? bili.scrollWidth > 520
+            ? `${(bili.scrollWidth * 9) / 16 + 70}px`
```

Comment:
This is actually not I am expecting, because there are videos which has other ratio.

A better api is providing a ratio option, and it has higher priority comparing with height.

Reply:
I used to use an array to configure the height, you can refer to [here](https://www.u2sb.com/OpenSw/vuepress-plugin-smplayer/bilibili.html#%E5%8F%82%E6%95%B0) for details, but if you do this, the changes will be relatively large, which is a destructive upgrade

Reply:
can like this 
```ts
height: {
  type: Array as PropType<Array<number>>,
  default: () => [9/16, 70],
  required: false,
}
```

```ts
height = scrollWidth * this.height[0] + this.height[1] + "px"
```